File: algorithms/easy/1_bit_and_2_bit_characters/solution.go
package main

import "fmt"

func isOneBitCharacter(bits []int) bool {
	skip := false
	for i := 0; i < len(bits)-1; i++ {
		if skip {
			skip = false
			continue
		}

		if bits[i] == 1 {
			skip = true
			continue
		}
	}

	if skip {
		return false
	}

	return true
}

func main() {
	fmt.Println(isOneBitCharacter([]int{1, 0, 0}) == true)
	fmt.Println(isOneBitCharacter([]int{1, 1, 1, 0}) == false)
}

File: algorithms/easy/1_bit_and_2_bit_characters/solution.py
class Solution:
    def isOneBitCharacter(self, bits: list) -> bool:
        skip = False
        for i in range(0, len(bits)-1):
            if skip:
                skip = False
                continue

            if bits[i] == 1:
                skip = True
                continue

        if skip:
            return False

        return True

solution = Solution()
print(solution.isOneBitCharacter([1,0,0])==True)
print(solution.isOneBitCharacter([1,1,1,0])==False)